pkg/prometheus/promcfg.go

```diff
@@ -322,14 +322,29 @@ func honorLabels(userHonorLabels, overrideHonorLabels bool) bool {
 	return userHonorLabels
 }
 
+// honorTimestamps determines the value of honor_timestamps.
+// if overrideHonorTimestamps is true, then we want to always
+// set honor_timestamps to true. Otherwise we set it to whateever
+// user wants, while assuming no setting as true.
+func honorTimestamps(userHonorTimestamps *bool, overrideHonorTimestamps bool) bool {
+	honor := true
+	if userHonorTimestamps != nil {
+		honor = *userHonorTimestamps
+	}
+
+	return honor && !overrideHonorTimestamps
+}
+
 func (cg *configGenerator) generatePodMonitorConfig(version semver.Version,
```

Comment:
this is not really Go style for large number args, but if that's kept in this codebase, then ok (: 

Should be:
```
function(
  arg1,
  arg2,
  ...
  argZ,
)
```

Reply:
Agreed, but it is also not in scope of this PR :)

Reply:
fixed.